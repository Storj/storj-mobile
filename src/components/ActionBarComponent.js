import {
    View,
    Text,
    StyleSheet,
    Image,
    TouchableOpacity
} from 'react-native';
import React, { Component } from 'react';
import { getWidth, getHeight } from '../utils/adaptive';
import TabBarActionModel from '../models/TabBarActionModel';
import PropTypes from 'prop-types';

/**
* pop up that is shown after pressing '+' button in TapBarComponent 
*/
export default class ActionBarComponent extends Component {
    constructor(props) {
        super(props);
    };

    deleteBuckets() {
        this.props.selectedBuckets.map(item => {
            this.props.deleteBucket(item);
        });
    };

    render() {
        return(
            <View style = { this.props.isSelectionMode ? styles.popUpRectangleWrapperSelectionMode : styles.popUpRectangleWrapper }>
                <View style = { styles.popUpRectangle }>
                    {
                            this.props.tapBarActions.map((action) => {
                                var icon = action.iconPath;
                                return (
                                    <View style = { styles.imageWrapper }>
                                        <TouchableOpacity onPress={ action.callback }> 
                                            <Image style = { styles.image } source = { action.icon }/>
                                        </TouchableOpacity>
                                    </View>);            
                            })
                        }
                </View>
                <View style = { styles.bottomTriangle }>
                    <Image source = { require('../images/ActionBar/ActionBarBottomTriangle.png') } style = { styles.triangleImage }/>
                </View>
            </View>
        );
    };
} 

ActionBarComponent.PropTypes = {
    isSelectionMode: PropTypes.bool,
    tapBarActions: PropTypes.arrayOf(PropTypes.instanceOf(TabBarActionModel))
};

const styles = StyleSheet.create({
    popUpRectangleWrapper: {
        position: 'absolute',
        bottom: getHeight(62),
        alignItems: 'center',
        marginHorizontal: getWidth(10)
    },  
    popUpRectangleWrapperSelectionMode: {
        position: 'absolute',
        bottom: getHeight(7),
        alignItems: 'center',
        marginHorizontal: getWidth(10)
    },  
    popUpRectangle: {
        width: getWidth(350),
        height: getHeight(51),
        borderRadius: getWidth(10),
        backgroundColor: '#2684FF',
        flexDirection: 'row'
    },
    imageWrapper: {
        width: getWidth(117),
        height: getHeight(50),
        paddingHorizontal: getWidth(46),
        paddingVertical: getHeight(10)
    },
    image: {
        width: getHeight(30),
        height: getHeight(30)
    },
    bottomTriangle: {
        height: getHeight(12),
        width: getWidth(350),
        alignItems: 'center'
    }
});