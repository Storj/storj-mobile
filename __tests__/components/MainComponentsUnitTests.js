import {
    View
} from 'react-native';
import React from 'react';
import Enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import MainComponent from '../../src/components/MainComponent';
import ActionBarComponent from '../../src/components/ActionBarComponent';
import CreateBucketPopUpComponent from '../../src/components/InputPopUpComponent';

Enzyme.configure({ adapter: new Adapter() });

describe('MainComponent', () => {    

	it('renders correctly', () => {

		const wrapper = shallow(
			<MainComponent />
        );
        
		expect(wrapper).toMatchSnapshot();
    });

    it('renders correctly with ActionBarShown', () => {

		const wrapper = shallow(
			<MainComponent 
                isActionBarShown = { true } />
        );

        expect(wrapper.containsMatchingElement(<ActionBarComponent />)).toBe(true);
        expect(wrapper.containsMatchingElement(<CreateBucketPopUpComponent />)).toBe(false);
        expect(wrapper).toMatchSnapshot();
    });

    it('renders correctly with ActionBarShown', () => {

		const wrapper = shallow(
			<MainComponent 
                isSelectionMode = { true } />
        );

        expect(wrapper.containsMatchingElement(<ActionBarComponent />)).toBe(true);
        expect(wrapper.containsMatchingElement(<CreateBucketPopUpComponent />)).toBe(false);
        expect(wrapper).toMatchSnapshot();
    });

    it('renders correctly with ActionBarShown', () => {

		const wrapper = shallow(
			<MainComponent 
                isCreateBucketInputShown = { true } />
        );

        expect(wrapper.containsMatchingElement(<ActionBarComponent />)).toBe(false);
        expect(wrapper.containsMatchingElement(<CreateBucketPopUpComponent />)).toBe(true);
        expect(wrapper).toMatchSnapshot();
    });

    it('renders correctly with ActionBarShown', () => {

		const wrapper = shallow(
			<MainComponent 
                isCreateBucketInputShown = { true } 
                isActionBarShown = { true } />
        );

        expect(wrapper.containsMatchingElement(<ActionBarComponent />)).toBe(true);
        expect(wrapper.containsMatchingElement(<CreateBucketPopUpComponent />)).toBe(true);
        expect(wrapper).toMatchSnapshot();
    });
});
