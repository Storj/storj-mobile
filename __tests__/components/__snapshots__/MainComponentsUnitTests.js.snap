// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MainComponent renders correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <MainComponent />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        <View
          style={
            Object {
              "flex": 1,
            }
          }
        >
          <Connect(MainNavigationContainer)
            isActionBarShown={undefined}
            isSelectionMode={undefined}
            isSingleItemSelected={undefined}
            onActionBarPress={[Function]}
          />
        </View>,
        null,
        null,
      ],
      "style": Object {
        "flex": 1,
      },
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": <Connect(MainNavigationContainer)
            isActionBarShown={undefined}
            isSelectionMode={undefined}
            isSingleItemSelected={undefined}
            onActionBarPress={[Function]}
          />,
          "style": Object {
            "flex": 1,
          },
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "isActionBarShown": undefined,
            "isSelectionMode": undefined,
            "isSingleItemSelected": undefined,
            "onActionBarPress": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": [Function],
      },
      null,
      null,
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <View
            style={
              Object {
                "flex": 1,
              }
            }
          >
            <Connect(MainNavigationContainer)
              isActionBarShown={undefined}
              isSelectionMode={undefined}
              isSingleItemSelected={undefined}
              onActionBarPress={[Function]}
            />
          </View>,
          null,
          null,
        ],
        "style": Object {
          "flex": 1,
        },
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": <Connect(MainNavigationContainer)
              isActionBarShown={undefined}
              isSelectionMode={undefined}
              isSingleItemSelected={undefined}
              onActionBarPress={[Function]}
            />,
            "style": Object {
              "flex": 1,
            },
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "isActionBarShown": undefined,
              "isSelectionMode": undefined,
              "isSingleItemSelected": undefined,
              "onActionBarPress": [Function],
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": [Function],
        },
        null,
        null,
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`MainComponent renders correctly with ActionBarShown 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <MainComponent
    isActionBarShown={true}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        <View
          style={
            Object {
              "flex": 1,
            }
          }
        >
          <Connect(MainNavigationContainer)
            isActionBarShown={true}
            isSelectionMode={undefined}
            isSingleItemSelected={undefined}
            onActionBarPress={[Function]}
          />
        </View>,
        <ActionBarComponent
          isSelectionMode={undefined}
          isSingleItemSelected={undefined}
          selectionModeActions={undefined}
          tapBarActions={undefined}
        />,
        null,
      ],
      "style": Object {
        "flex": 1,
      },
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": <Connect(MainNavigationContainer)
            isActionBarShown={true}
            isSelectionMode={undefined}
            isSingleItemSelected={undefined}
            onActionBarPress={[Function]}
          />,
          "style": Object {
            "flex": 1,
          },
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "isActionBarShown": true,
            "isSelectionMode": undefined,
            "isSingleItemSelected": undefined,
            "onActionBarPress": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "isSelectionMode": undefined,
          "isSingleItemSelected": undefined,
          "selectionModeActions": undefined,
          "tapBarActions": undefined,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      null,
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <View
            style={
              Object {
                "flex": 1,
              }
            }
          >
            <Connect(MainNavigationContainer)
              isActionBarShown={true}
              isSelectionMode={undefined}
              isSingleItemSelected={undefined}
              onActionBarPress={[Function]}
            />
          </View>,
          <ActionBarComponent
            isSelectionMode={undefined}
            isSingleItemSelected={undefined}
            selectionModeActions={undefined}
            tapBarActions={undefined}
          />,
          null,
        ],
        "style": Object {
          "flex": 1,
        },
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": <Connect(MainNavigationContainer)
              isActionBarShown={true}
              isSelectionMode={undefined}
              isSingleItemSelected={undefined}
              onActionBarPress={[Function]}
            />,
            "style": Object {
              "flex": 1,
            },
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "isActionBarShown": true,
              "isSelectionMode": undefined,
              "isSingleItemSelected": undefined,
              "onActionBarPress": [Function],
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "isSelectionMode": undefined,
            "isSingleItemSelected": undefined,
            "selectionModeActions": undefined,
            "tapBarActions": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        null,
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`MainComponent renders correctly with ActionBarShown 2`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <MainComponent
    isSelectionMode={true}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        <View
          style={
            Object {
              "flex": 1,
            }
          }
        >
          <Connect(MainNavigationContainer)
            isActionBarShown={undefined}
            isSelectionMode={true}
            isSingleItemSelected={undefined}
            onActionBarPress={[Function]}
          />
        </View>,
        <ActionBarComponent
          isSelectionMode={true}
          isSingleItemSelected={undefined}
          selectionModeActions={undefined}
          tapBarActions={undefined}
        />,
        null,
      ],
      "style": Object {
        "flex": 1,
      },
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": <Connect(MainNavigationContainer)
            isActionBarShown={undefined}
            isSelectionMode={true}
            isSingleItemSelected={undefined}
            onActionBarPress={[Function]}
          />,
          "style": Object {
            "flex": 1,
          },
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "isActionBarShown": undefined,
            "isSelectionMode": true,
            "isSingleItemSelected": undefined,
            "onActionBarPress": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "isSelectionMode": true,
          "isSingleItemSelected": undefined,
          "selectionModeActions": undefined,
          "tapBarActions": undefined,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      null,
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <View
            style={
              Object {
                "flex": 1,
              }
            }
          >
            <Connect(MainNavigationContainer)
              isActionBarShown={undefined}
              isSelectionMode={true}
              isSingleItemSelected={undefined}
              onActionBarPress={[Function]}
            />
          </View>,
          <ActionBarComponent
            isSelectionMode={true}
            isSingleItemSelected={undefined}
            selectionModeActions={undefined}
            tapBarActions={undefined}
          />,
          null,
        ],
        "style": Object {
          "flex": 1,
        },
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": <Connect(MainNavigationContainer)
              isActionBarShown={undefined}
              isSelectionMode={true}
              isSingleItemSelected={undefined}
              onActionBarPress={[Function]}
            />,
            "style": Object {
              "flex": 1,
            },
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "isActionBarShown": undefined,
              "isSelectionMode": true,
              "isSingleItemSelected": undefined,
              "onActionBarPress": [Function],
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "isSelectionMode": true,
            "isSingleItemSelected": undefined,
            "selectionModeActions": undefined,
            "tapBarActions": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        null,
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`MainComponent renders correctly with ActionBarShown 3`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <MainComponent
    isCreateBucketInputShown={true}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        <View
          style={
            Object {
              "flex": 1,
            }
          }
        >
          <Connect(MainNavigationContainer)
            isActionBarShown={undefined}
            isSelectionMode={undefined}
            isSingleItemSelected={undefined}
            onActionBarPress={[Function]}
          />
        </View>,
        null,
        <View
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "transparent",
              "bottom": 0,
              "justifyContent": "center",
              "left": 0,
              "position": "absolute",
              "right": 0,
              "top": 0,
            }
          }
        >
          <View
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "backgroundColor": "transparent",
                  "bottom": 0,
                  "justifyContent": "center",
                  "left": 0,
                  "position": "absolute",
                  "right": 0,
                  "top": 0,
                },
                Object {
                  "backgroundColor": "black",
                  "opacity": 0.4,
                },
              ]
            }
          />
          <CreateBucketPopUpComponent
            onApply={[Function]}
            onCancel={[Function]}
          />
        </View>,
      ],
      "style": Object {
        "flex": 1,
      },
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": <Connect(MainNavigationContainer)
            isActionBarShown={undefined}
            isSelectionMode={undefined}
            isSingleItemSelected={undefined}
            onActionBarPress={[Function]}
          />,
          "style": Object {
            "flex": 1,
          },
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "isActionBarShown": undefined,
            "isSelectionMode": undefined,
            "isSingleItemSelected": undefined,
            "onActionBarPress": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": [Function],
      },
      null,
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <View
              style={
                Array [
                  Object {
                    "alignItems": "center",
                    "backgroundColor": "transparent",
                    "bottom": 0,
                    "justifyContent": "center",
                    "left": 0,
                    "position": "absolute",
                    "right": 0,
                    "top": 0,
                  },
                  Object {
                    "backgroundColor": "black",
                    "opacity": 0.4,
                  },
                ]
              }
            />,
            <CreateBucketPopUpComponent
              onApply={[Function]}
              onCancel={[Function]}
            />,
          ],
          "style": Object {
            "alignItems": "center",
            "backgroundColor": "transparent",
            "bottom": 0,
            "justifyContent": "center",
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
          },
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "style": Array [
                Object {
                  "alignItems": "center",
                  "backgroundColor": "transparent",
                  "bottom": 0,
                  "justifyContent": "center",
                  "left": 0,
                  "position": "absolute",
                  "right": 0,
                  "top": 0,
                },
                Object {
                  "backgroundColor": "black",
                  "opacity": 0.4,
                },
              ],
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "onApply": [Function],
              "onCancel": [Function],
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <View
            style={
              Object {
                "flex": 1,
              }
            }
          >
            <Connect(MainNavigationContainer)
              isActionBarShown={undefined}
              isSelectionMode={undefined}
              isSingleItemSelected={undefined}
              onActionBarPress={[Function]}
            />
          </View>,
          null,
          <View
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "transparent",
                "bottom": 0,
                "justifyContent": "center",
                "left": 0,
                "position": "absolute",
                "right": 0,
                "top": 0,
              }
            }
          >
            <View
              style={
                Array [
                  Object {
                    "alignItems": "center",
                    "backgroundColor": "transparent",
                    "bottom": 0,
                    "justifyContent": "center",
                    "left": 0,
                    "position": "absolute",
                    "right": 0,
                    "top": 0,
                  },
                  Object {
                    "backgroundColor": "black",
                    "opacity": 0.4,
                  },
                ]
              }
            />
            <CreateBucketPopUpComponent
              onApply={[Function]}
              onCancel={[Function]}
            />
          </View>,
        ],
        "style": Object {
          "flex": 1,
        },
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": <Connect(MainNavigationContainer)
              isActionBarShown={undefined}
              isSelectionMode={undefined}
              isSingleItemSelected={undefined}
              onActionBarPress={[Function]}
            />,
            "style": Object {
              "flex": 1,
            },
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "isActionBarShown": undefined,
              "isSelectionMode": undefined,
              "isSingleItemSelected": undefined,
              "onActionBarPress": [Function],
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": [Function],
        },
        null,
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <View
                style={
                  Array [
                    Object {
                      "alignItems": "center",
                      "backgroundColor": "transparent",
                      "bottom": 0,
                      "justifyContent": "center",
                      "left": 0,
                      "position": "absolute",
                      "right": 0,
                      "top": 0,
                    },
                    Object {
                      "backgroundColor": "black",
                      "opacity": 0.4,
                    },
                  ]
                }
              />,
              <CreateBucketPopUpComponent
                onApply={[Function]}
                onCancel={[Function]}
              />,
            ],
            "style": Object {
              "alignItems": "center",
              "backgroundColor": "transparent",
              "bottom": 0,
              "justifyContent": "center",
              "left": 0,
              "position": "absolute",
              "right": 0,
              "top": 0,
            },
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "style": Array [
                  Object {
                    "alignItems": "center",
                    "backgroundColor": "transparent",
                    "bottom": 0,
                    "justifyContent": "center",
                    "left": 0,
                    "position": "absolute",
                    "right": 0,
                    "top": 0,
                  },
                  Object {
                    "backgroundColor": "black",
                    "opacity": 0.4,
                  },
                ],
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "onApply": [Function],
                "onCancel": [Function],
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`MainComponent renders correctly with ActionBarShown 4`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <MainComponent
    isActionBarShown={true}
    isCreateBucketInputShown={true}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        <View
          style={
            Object {
              "flex": 1,
            }
          }
        >
          <Connect(MainNavigationContainer)
            isActionBarShown={true}
            isSelectionMode={undefined}
            isSingleItemSelected={undefined}
            onActionBarPress={[Function]}
          />
        </View>,
        <ActionBarComponent
          isSelectionMode={undefined}
          isSingleItemSelected={undefined}
          selectionModeActions={undefined}
          tapBarActions={undefined}
        />,
        <View
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "transparent",
              "bottom": 0,
              "justifyContent": "center",
              "left": 0,
              "position": "absolute",
              "right": 0,
              "top": 0,
            }
          }
        >
          <View
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "backgroundColor": "transparent",
                  "bottom": 0,
                  "justifyContent": "center",
                  "left": 0,
                  "position": "absolute",
                  "right": 0,
                  "top": 0,
                },
                Object {
                  "backgroundColor": "black",
                  "opacity": 0.4,
                },
              ]
            }
          />
          <CreateBucketPopUpComponent
            onApply={[Function]}
            onCancel={[Function]}
          />
        </View>,
      ],
      "style": Object {
        "flex": 1,
      },
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": <Connect(MainNavigationContainer)
            isActionBarShown={true}
            isSelectionMode={undefined}
            isSingleItemSelected={undefined}
            onActionBarPress={[Function]}
          />,
          "style": Object {
            "flex": 1,
          },
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "isActionBarShown": true,
            "isSelectionMode": undefined,
            "isSingleItemSelected": undefined,
            "onActionBarPress": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "isSelectionMode": undefined,
          "isSingleItemSelected": undefined,
          "selectionModeActions": undefined,
          "tapBarActions": undefined,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <View
              style={
                Array [
                  Object {
                    "alignItems": "center",
                    "backgroundColor": "transparent",
                    "bottom": 0,
                    "justifyContent": "center",
                    "left": 0,
                    "position": "absolute",
                    "right": 0,
                    "top": 0,
                  },
                  Object {
                    "backgroundColor": "black",
                    "opacity": 0.4,
                  },
                ]
              }
            />,
            <CreateBucketPopUpComponent
              onApply={[Function]}
              onCancel={[Function]}
            />,
          ],
          "style": Object {
            "alignItems": "center",
            "backgroundColor": "transparent",
            "bottom": 0,
            "justifyContent": "center",
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
          },
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "style": Array [
                Object {
                  "alignItems": "center",
                  "backgroundColor": "transparent",
                  "bottom": 0,
                  "justifyContent": "center",
                  "left": 0,
                  "position": "absolute",
                  "right": 0,
                  "top": 0,
                },
                Object {
                  "backgroundColor": "black",
                  "opacity": 0.4,
                },
              ],
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "onApply": [Function],
              "onCancel": [Function],
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <View
            style={
              Object {
                "flex": 1,
              }
            }
          >
            <Connect(MainNavigationContainer)
              isActionBarShown={true}
              isSelectionMode={undefined}
              isSingleItemSelected={undefined}
              onActionBarPress={[Function]}
            />
          </View>,
          <ActionBarComponent
            isSelectionMode={undefined}
            isSingleItemSelected={undefined}
            selectionModeActions={undefined}
            tapBarActions={undefined}
          />,
          <View
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "transparent",
                "bottom": 0,
                "justifyContent": "center",
                "left": 0,
                "position": "absolute",
                "right": 0,
                "top": 0,
              }
            }
          >
            <View
              style={
                Array [
                  Object {
                    "alignItems": "center",
                    "backgroundColor": "transparent",
                    "bottom": 0,
                    "justifyContent": "center",
                    "left": 0,
                    "position": "absolute",
                    "right": 0,
                    "top": 0,
                  },
                  Object {
                    "backgroundColor": "black",
                    "opacity": 0.4,
                  },
                ]
              }
            />
            <CreateBucketPopUpComponent
              onApply={[Function]}
              onCancel={[Function]}
            />
          </View>,
        ],
        "style": Object {
          "flex": 1,
        },
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": <Connect(MainNavigationContainer)
              isActionBarShown={true}
              isSelectionMode={undefined}
              isSingleItemSelected={undefined}
              onActionBarPress={[Function]}
            />,
            "style": Object {
              "flex": 1,
            },
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "isActionBarShown": true,
              "isSelectionMode": undefined,
              "isSingleItemSelected": undefined,
              "onActionBarPress": [Function],
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "isSelectionMode": undefined,
            "isSingleItemSelected": undefined,
            "selectionModeActions": undefined,
            "tapBarActions": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <View
                style={
                  Array [
                    Object {
                      "alignItems": "center",
                      "backgroundColor": "transparent",
                      "bottom": 0,
                      "justifyContent": "center",
                      "left": 0,
                      "position": "absolute",
                      "right": 0,
                      "top": 0,
                    },
                    Object {
                      "backgroundColor": "black",
                      "opacity": 0.4,
                    },
                  ]
                }
              />,
              <CreateBucketPopUpComponent
                onApply={[Function]}
                onCancel={[Function]}
              />,
            ],
            "style": Object {
              "alignItems": "center",
              "backgroundColor": "transparent",
              "bottom": 0,
              "justifyContent": "center",
              "left": 0,
              "position": "absolute",
              "right": 0,
              "top": 0,
            },
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "style": Array [
                  Object {
                    "alignItems": "center",
                    "backgroundColor": "transparent",
                    "bottom": 0,
                    "justifyContent": "center",
                    "left": 0,
                    "position": "absolute",
                    "right": 0,
                    "top": 0,
                  },
                  Object {
                    "backgroundColor": "black",
                    "opacity": 0.4,
                  },
                ],
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "onApply": [Function],
                "onCancel": [Function],
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
